src/hotspot/share/gc/g1/g1HeapRegion.cpp

```diff
@@ -336,7 +336,7 @@ class VerifyCodeRootCodeBlobClosure: public CodeBlobClosure {
     _hr(hr), _failures(false) {}
 
   void do_code_blob(CodeBlob* cb) {
-    nmethod* nm = (cb == nullptr) ? nullptr : cb->as_compiled_method()->as_nmethod_or_null();
+    nmethod* nm = (cb == nullptr) ? nullptr : cb->as_nmethod_or_null();
```

Comment:
After this change I'd like to see if we can change this and similar GC interfaces to work directly against `nmethod` instead of `CodeBlob`.